pex/pep_508.py

```diff
@@ -0,0 +1,157 @@
+# Copyright 2022 Pants project contributors (see CONTRIBUTORS.md).
```

Comment:
Given that the class is called `MarkerEnvironment`, how come you didn't go with `marker_environment.py`?

Reply:
In my head - markers == PEP-508. It's true its just a subset of that  PEP though.